test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/TestFiles/Output/DesignTime/Basic.cs

```diff
@@ -1,47 +1,110 @@
 namespace AspNetCore
 {
+    #line hidden
+    using System;
     using System.Threading.Tasks;
+#line 2 ""
+using System.Linq;
 
-    public class testfiles_input_basic_cshtml : Microsoft.AspNetCore.Mvc.Razor.RazorPage<dynamic>
+#line default
+#line hidden
+#line 3 ""
+using System.Collections.Generic;
+
+#line default
+#line hidden
+#line 4 ""
+using Microsoft.AspNetCore.Mvc;
+
+#line default
+#line hidden
+#line 5 ""
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+#line default
+#line hidden
+#line 6 ""
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+
+#line default
+#line hidden
+    public class TestFiles_Input_Basic_cshtml : global::Microsoft.AspNetCore.Mvc.Razor.RazorPage<dynamic>
     {
-        private static object @__o;
-        private void @__RazorDesignTimeHelpers__()
-        {
-            #pragma warning disable 219
-            #pragma warning restore 219
+        #pragma warning disable 219
+        private void __RazorDirectiveTokenHelpers__() {
+        ((System.Action)(() => {
+global::Microsoft.AspNetCore.Mvc.Rendering.IHtmlHelper<TModel> __typeHelper = null;
         }
-        #line hidden
-        public testfiles_input_basic_cshtml()
-        {
+        ))();
+        ((System.Action)(() => {
+System.Object Html = null;
         }
-        #line hidden
-        [Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]
-        public Microsoft.AspNetCore.Mvc.ViewFeatures.IModelExpressionProvider ModelExpressionProvider { get; private set; }
-        [Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]
-        public Microsoft.AspNetCore.Mvc.IUrlHelper Url { get; private set; }
-        [Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]
-        public Microsoft.AspNetCore.Mvc.IViewComponentHelper Component { get; private set; }
-        [Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]
-        public Microsoft.AspNetCore.Mvc.Rendering.IJsonHelper Json { get; private set; }
-        [Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]
-        public Microsoft.AspNetCore.Mvc.Rendering.IHtmlHelper<dynamic> Html { get; private set; }
-
-        #line hidden
-
+        ))();
+        ((System.Action)(() => {
+global::Microsoft.AspNetCore.Mvc.Rendering.IJsonHelper __typeHelper = null;
+        }
+        ))();
+        ((System.Action)(() => {
+System.Object Json = null;
+        }
+        ))();
+        ((System.Action)(() => {
+global::Microsoft.AspNetCore.Mvc.IViewComponentHelper __typeHelper = null;
+        }
+        ))();
+        ((System.Action)(() => {
+System.Object Component = null;
+        }
+        ))();
+        ((System.Action)(() => {
+global::Microsoft.AspNetCore.Mvc.IUrlHelper __typeHelper = null;
+        }
+        ))();
+        ((System.Action)(() => {
+System.Object Url = null;
+        }
+        ))();
+        ((System.Action)(() => {
```

Comment:
This is part of the new designtime codegen coming from extensible directives. We generate placeholders so we can add line mappings to the things that you put in directives. Basically this gives us the ability to colourize the type name in `@inject`.